proposals/0055-transact-without-require-weight-at-most.md

```diff
@@ -0,0 +1,65 @@
+---
+Title: Transact without `require_weight_at_most`
+Number: 55
+Status: Draft
+Version: 0
+Authors:
+ - Adrian Catangiu
+ - Francisco Aguirre
+Created: 2024-04-25
+Impact: High
+Requires:
+Replaces:
+---
+
+## Summary
+
+The Transact instruction shouldn't need the `require_weight_at_most` parameter, since the decoder could just get the weight of the call from its dispatch info.
+
+## Motivation
+
+The UX of using `Transact` is not great, and one part of the problem is guesstimating this `require_weight_at_most`.
+We've seen multiple `Transact`s on-chain failures caused by the "incorrect" use or the parameter. In practice,
+this parameter only adds UX overhead. The vast majority of use cases fall in one of two categories:
+1. Unpaid execution of Transacts - in these cases the `require_weight_at_most` is not really useful, caller doesn't 
+have to pay for it, and on the call site it either fits the block or not;
+2. Paid execution of _single_ Transact - the weight to be spent by the Transact is already covered by the `BuyExecution`
+weight limit parameter.
+
+We've had multiple OpenGov `root/whitelisted_caller` proposals initiated by core-devs completely or partially fail
+because of incorrect configuration of `require_weight_at_most` parameter. This is a strong indication that the
+instruction is hard to use.
+
+## Specification
+
+The proposal is simple: change `Transact` instruction:
+
+```diff
+- Transact { origin_kind: OriginKind, require_weight_at_most: Weight, call: DoubleEncoded<Call> },
++ Transact { origin_kind: OriginKind, maybe_weight_limit: Option<WeightLimit>, call: DoubleEncoded<Call> },
```

Comment:
Since `WeightLimit` [has the `Unlimited` variant](https://github.com/paritytech/polkadot-sdk/blob/ad22fa6e785bccfb5e1ceb113c870e02f132ce46/polkadot/xcm/src/v3/mod.rs#L434-L436), wrapping it into `Option` seems unnecessary unless there are plans to change this type

Reply:
Opened https://github.com/paritytech/xcm-format/pull/61 to clarify WeightLimit